apps/framework-cli/src/mcp/tools/toon_serializer.rs

```diff
@@ -0,0 +1,166 @@
+//! TOON format serialization utilities for MCP tool responses
+//!
+//! This module provides functions to serialize JSON data to TOON format,
+//! optimized for compression to reduce token usage when transmitting data
+//! to Large Language Models through the Model Context Protocol.
+
+use serde_json::Value;
+use toon_format::types::KeyFoldingMode;
+use toon_format::{encode, encode_default, EncodeOptions};
+
+/// Errors that can occur during TOON serialization
+#[derive(Debug, thiserror::Error)]
+pub enum ToonSerializationError {
+    #[error("failed to encode data to TOON format: {message}")]
+    EncodingError { message: String },
+}
+
+/// Serialize a JSON value to TOON format with compression optimization
+///
+/// Uses key folding to collapse nested object chains for better compression.
+/// This is the recommended encoding method for large data payloads.
+///
+/// # Arguments
+/// * `value` - The JSON value to serialize
+///
+/// # Returns
+/// * `Ok(String)` - TOON-formatted string
+/// * `Err(ToonSerializationError)` - If encoding fails
+///
+/// # Example
+/// ```rust
+/// let data = json!({"users": [{"id": 1, "name": "Alice"}]});
+/// let toon_str = serialize_to_toon_compressed(&data)?;
+/// ```
+pub fn serialize_to_toon_compressed(value: &Value) -> Result<String, ToonSerializationError> {
```

Comment:
Is this old? Your issue got fixed for this https://github.com/toon-format/toon-rust/issues/21